modules/tinymce/src/core/main/ts/api/commands/SelectionCommands.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import * as NodeType from '../../dom/NodeType';
+import type Editor from '../Editor';
+
+export const registerCommands = (editor: Editor) => {
+  editor.editorCommands.addCommands({
+    mceSelectNodeDepth: (_command, _ui, value) => {
+      let counter = 0;
+
+      editor.dom.getParent(editor.selection.getNode(), (node) => {
+        if (node.nodeType === 1 && counter++ === value) {
```

Comment:
Not something to do in this PR, but one day we should clean up all these magic numbers and make them use the relevant helper functions or constants we have already.

Reply:
Yes, I think predicates for all these would be better since it can then do things like `is Element` etc on types not sure it can do that by property inspection.